tests/test_external_search.py

```diff
@@ -396,490 +453,275 @@ def test_query_works(self):
+        # A query without an apostrophe matches a word that contains one.
+        # (NOTE: it's not clear whether this is a feature of the index or
+        # done by the fuzzy match.)
+        expect(self.tess, "durbervilles")
+        expect(self.tiffany, "tiffanys")
 
+        # A query with an 'e' matches a word that contains an
+        # e-with-acute. (NOTE: it's not clear whether this is a
+        # feature of the index or done by the fuzzy match.)
+        expect(self.les_mis, "les miserables")
 
-        # Ranks closest target age range highest
+        # Find results based on fiction status.
+        #
+        # Here, Moby-Dick (fiction) is privileged over Moby Duck
+        # (nonfiction)
+        expect([self.moby_dick, self.moby_duck], "fiction moby")
 
-        results = query("age 3-5", None, None, None, None, None, None, None)
-        hits = results["hits"]["hits"]
-        eq_(3, len(hits))
-        eq_(unicode(self.age_4_5.id), hits[0]['_id'])
-        eq_(unicode(self.age_5_6.id), hits[1]['_id'])
-        eq_(unicode(self.age_2_10.id), hits[2]['_id'])
+        # Here, Moby Duck is privileged over Moby-Dick.
+        expect([self.moby_duck, self.moby_dick], "nonfiction moby")
 
+        # Find results based on genre.
 
-        # Matches genre + audience
+        # The name of the genre also shows up in the title of a book,
+        # but the genre boost means the romance novel is the first
+        # result.
+        expect([self.ya_romance, self.modern_romance], "romance")
 
-        results = query("young adult romance", None, None, None, None, None, None, None)
-        hits = results["hits"]["hits"]
+        # Find results based on audience.
+        expect(self.children_work, "children's")
 
-        # The book with 'Romance' in the title also shows up, but it
-        # shows up after the book whose audience matches 'young adult'
-        # and whose genre matches 'romance'.
-        eq_(
-            map(unicode, [self.ya_romance.id, self.modern_romance.id]),
-            [x['_id'] for x in hits]
+        expect(
+            [self.ya_work, self.ya_romance], "young adult", ordered=False
         )
 
-        # Matches age + fiction
-
-        results = query("age 5 fiction", None, None, None, None, None, None, None)
-        hits = results["hits"]["hits"]
-        eq_(2, len(hits))
-        eq_(unicode(self.age_4_5.id), hits[0]['_id'])
+        # Find results based on grade level or target age.
+        for q in ('grade 4', 'grade 4-6', 'age 9'):
+            # ages 9-10 is a better result because a book targeted
+            # toward a narrow range is a better match than a book
+            # targeted toward a wide range.
+            expect([self.age_9_10, self.age_2_10], q)
 
+        # TODO: I expected this to act like the other queries, but the
+        # 2-10 book shows up above the 9-10 book. This might indicate
+        # a bug.
+        expect([self.age_9_10, self.age_2_10], "age 10-12", ordered=False)
```

Comment:
I'm pretty sure this behavior is not a change from the previous behavior, but not 100%.

Reply:
I don't think it's a change. The target age query only scores how big the overlap is, it doesn't look at how large the non-overlapping part of the range is. That could be a good thing to add.